bundles/framework/layerlist/Flyout.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { LayerList } from './view/LayerList';
+/**
+ * @class Oskari.mapframework.bundle.layerlist.Flyout
+ *
+ * Renders the "all layers" flyout.
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.layerlist.Flyout',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     * @param {Oskari.mapframework.bundle.layerlist.LayerListBundleInstance} instance
+     *    reference to component that created the flyout
+     */
+    function (instance) {
+        this.instance = instance;
+        this.container = null;
+    }, {
+
+        /**
+         * @method getName
+         * @return {String} the name for the component
+         */
+        getName: function () {
+            return 'Oskari.mapframework.bundle.layerlist.Flyout';
+        },
+
+        /**
+         * @method setEl
+         * @param {Object} el
+         *     reference to the container in browser
+         * @param {Number} width
+         *     container size(?) - not used
+         * @param {Number} height
+         *     container size(?) - not used
+         *
+         * Interface method implementation
+         */
+        setEl: function (el, flyout, width, height) {
+            this.container = el[0];
+            if (!jQuery(this.container).hasClass('layerlist')) {
+                jQuery(this.container).addClass('layerlist');
+            }
+            if (!flyout.hasClass('layerlist')) {
+                flyout.addClass('layerlist');
+            }
+        },
+
+        /**
+         * @method startPlugin
+         *
+         * Interface method implementation, assigns the HTML templates that will be used to create the UI
+         */
+        startPlugin: function () {
+            var elParent = this.container.parentElement.parentElement;
+            var elId = jQuery(elParent).find('.oskari-flyouttoolbar .oskari-flyouttools .oskari-flyouttool-close');
+            elId.attr('id', 'oskari_layerselector2_flyout_oskari_flyouttool_close');
```

Comment:
I think the ID is for Selenium testing so lets remove it for now. We can use the new t_something class for this if it's still required by tests but as this is a new bundle it's perfect time to correct previous sins.